GitCommands/Git/Extensions/ProcessExtensions.cs

```diff
@@ -1,40 +1,70 @@
 ﻿using System.Diagnostics;
 using System.Runtime.InteropServices;
 using GitCommands.Utils;
+using GitUIPluginInterfaces;
 
 namespace GitCommands.Git.Extensions
 {
     public static class ProcessExtensions
     {
-        public static void TerminateTree(this Process process)
+        /// <summary>
+        /// Send Ctrl+C to the process.
+        /// </summary>
+        public static bool SendTerminationRequest(this Process process)
         {
-            if (EnvUtils.RunningOnWindows())
+            if (!EnvUtils.RunningOnWindows())
+            {
+                return false;
+            }
+
+            if (!NativeMethods.AttachConsole(process.Id) && Marshal.GetLastWin32Error() != NativeMethods.AlreadyAttachedToConsole)
+            {
+                return false;
+            }
+
+            try
+            {
+                _ = NativeMethods.SetConsoleCtrlHandler(IntPtr.Zero, add: true);
+
+                return NativeMethods.GenerateConsoleCtrlEvent(0, 0);
```

Comment:
Why <kbd>CTRL+C</kbd> and not <kbd>CTRL+BREAK</kbd>?
![image](https://github.com/gitextensions/gitextensions/assets/4403806/1a478b52-541b-43d4-b06c-e0d7625166b7)

Reply:
Ctrl+Break kills the current process, too. Then f.i. the test runner is killed and does not yield test results.
I tried to intercept it by setting a handler using `SetConsoleCtrlHandler`. The handler was not called.